### PR TITLE
axis change of make_leave_one_out

### PR DESCRIPTION
**Pull request by Gusanidas**
current implementation asumes the axis is going to be expressed as a negative number, and has different behaviour if its expressed as a positive number.

for example, if the input tensor shape is (5,4,6), the output with axis =1 and axis=-2 should have shape (5,4,3,6)

### CODE REVIEW

**Comment by clement-bonnet** [comment]
Thank you for your comment!
I think `axis=1` and `axis=-2` actually mean different things. E.g. think of the `out_axes` argument of `jax.vmap`, it indicates where the dimension should end up. So, with a tensor shape `(5,4,6)`, I would argue that the axis values that are equivalent are:
- `axis=0` and `axis=-4` -> `(x,5,4,6)`
- `axis=1` and `axis=-3` -> `(x,5,4,6)`
- `axis=2` and `axis=-2` -> `(x,5,4,6)`
- `axis=3` and `axis=-1` -> `(x,5,4,6)`
Which works with the current code I believe.

**Comment by Gusanidas** [comment]
I think you forgot to move the x in the examples?

axis=0 and axis=-4 -> (x,5,4,6)
axis=1 and axis=-3 -> (5,x,4,6)
axis=2 and axis=-2 -> (5,4,x,6)
axis=3 and axis=-1 -> (5,4,6,x)

But that is not what the current code does if the axis value is positive. For example if the original shape is:

(a,b,c,d) and the axis=1, the current code will return:

(b,a,b-1,c,d)

That I believe is not what the function is intended to do, and is not what the docstirng says.

With the change I made the behaviour if the axis is negative is the same. And the axis with one element left out is always the next to the one picked.

(a,b,c,d) and axis=1:

(a,b,b-1,c,d)

**Comment by Gusanidas** [comment]
pd: thanks for answering and taking the time to check my pull request

**Review by clement-bonnet — APPROVED** [review]
Right, thank you for spotting this!
